### ISSUE TITLE
Enable geographical coordinates for the --closest option of the GeoBase command-line

### ISSUE THREAD

**Issue by da115115** [opentraveldata/geobases#1]
Currently, the command-line does not fully allow to search the closest POR entries of a POR specified by its geographical coordinates. For instance the following works:

``` shell
$ GeoBase --closest \(-5, 140\)
```

while the following does not work:

``` shell
$ GeoBase --closest \(65, -14\)
GeoBase: error: unrecognized arguments: -14)
```

Note that the following works:

``` shell
$ GeoBase --closest "(65, -14)"
```

So, the enhancement could just be a documentation issue :)

**Comment by alexprengere** [comment]
Thanks for opening this issue! I checked, the error occurs whe `argparse` is parsing the command line.
The problem comes down to this: when you do not put quotes around your geocode, argparse will try to interpret any "-" as a new option. The `--closest` and `--near` options accept any number of arguments and expects one key or a geocode, but this stops when new options are encountered. In this error case, `-14` is indeed an unknown option, so argparse fails. 

The preferred way to give a geocode input to these options is to use quotes (the `--help` tells you that, but I will try to make that more obvious), and by the way you do not have to put the parenthesis (they are stripped in the code). If you really wish _not to_ put quotes around the geocode, you can still escape the "-", it will be removed by the shell:

```
$  GeoBase --closest 65, \-14     
```

To sum up, I will try to make the documentation clear on the fact that this is usual way to do it:

```
$  GeoBase --closest '65, -14'
```

**Comment by alexprengere** [comment]
Hopefully 5aaabf83b8dacefbb40ad8c161af37363303b485 and 1e18b1a30b562d06888a150854a71f3be14b7036 will avoid further confusion, updated the site as well.

**Comment by da115115** [comment]
That is already good. May I suggest that the option parser tool be a little bit more flexible, such as to allow the following forms (currently failing)?

```
$  GeoBase --closest '65 -14'
$  GeoBase --closest '65^-14'
$  GeoBase --closest '65;-14'
```

Also, interestingly, the following form succeeds:

```
$  GeoBase --closest \(-65, 14\)
```

Couldn't we trick the Python argument parser, for instance by specifying `-<numerical>`/`+<numerical>` options, where those numerical options would be interpreted in the context, that is, depending on the other options last parsed (e.g., `--closest` and `--near` in our case)?

**Comment by alexprengere** [comment]
2dee2a008810edd858c4a94da3a72643d0f96775 should make the parser more flexible, allowing the forms you mentionned (and allowing also spaces and tabs).

I think the succeeding form you mention works because the "-" does not follow a white space, so argparse does not consider this as a new option (it must be looking for the `[ ]+--?[:alnum:]` regexp for new options, something like that).

About parser flexibility, it is indeed possible to specify `type=float` for the `--closest` option, and it would interpret negative number correctly. _But_ then we would lose flexbility on all forms like `(48, 2)` or `48;2`, because these do not look like floats. So here are the choices: 
- `type=float` would only allow space-separated floats to describe a geocode, but would correctly interpret negative numbers without quotes
- `type=string` (current behavior) would fail on non-quoted negative numbers and that's it, once argparse does not fail, we control completely the geocode parsing.

I think the second solution is more flexible, since it is very easy to quote the geocode. Moreover, `--closest` also accepts simple keys as argument (like `ORY`), so in fact we do not really have a choice here, unless we create another option dedicated to geocode parsing, but this would be a bit of an overkill :).
